commands/service_delete_test.go

```diff
@@ -3,198 +3,65 @@ package commands
-
-	require.NoError(t, c.cmd.Execute())
-
-	sm.AssertExpectations(t)
-	me.AssertExpectations(t)
-}
-
-func TestServiceDeleteWithAllFlagWithServiceAndDataConfirmation(t *testing.T) {
-	var (
-		sm = newMockSurvey()
-		me = newMockExecutor()
-		c  = newServiceDeleteCmd(me, sm)
-	)
-
-	c.cmd.Flags().Set("all", "true")
-
-	sm.On("AskOne", &survey.Confirm{
-		Message: "Are you sure to delete all services?",
-		Default: false,
-	}, mock.Anything, mock.Anything).Once().Return(nil).Run(confirmSurvey())
-
-	sm.On("AskOne", &survey.Confirm{
-		Message: "Do you want to remove service(s)' persistent data as well?",
-		Default: false,
-	}, mock.Anything, mock.Anything).Once().Return(nil).Run(confirmSurvey())
 
-	me.On("ServiceDeleteAll", true).Once().Return(nil)
-
-	require.NoError(t, c.cmd.Execute())
-
-	sm.AssertExpectations(t)
-	me.AssertExpectations(t)
-}
-
-func TestServiceDeleteWithAllFlagWithoutDeleteServiceConfirmation(t *testing.T) {
-	var (
-		sm = newMockSurvey()
-		me = newMockExecutor()
-		c  = newServiceDeleteCmd(me, sm)
-	)
-
-	c.cmd.Flags().Set("all", "true")
-
-	sm.On("AskOne", &survey.Confirm{
-		Message: "Are you sure to delete all services?",
-		Default: false,
-	}, mock.Anything, mock.Anything).Once().Return(nil).Run(rejectSurvey())
-
-	require.Contains(t, errConfirmationNeeded.Error(), c.cmd.Execute().Error())
-
-	sm.AssertExpectations(t)
-	me.AssertExpectations(t)
-}
-
-func TestServiceDeleteWithServiceID(t *testing.T) {
-	var (
-		serviceID = "1"
-		sm        = newMockSurvey()
-		me        = newMockExecutor()
-		c         = newServiceDeleteCmd(me, sm)
-	)
-
-	c.cmd.SetArgs([]string{serviceID})
-
-	sm.On("AskOne", &survey.Confirm{
-		Message: "Are you sure to delete service(s)?",
-		Default: false,
-	}, mock.Anything, mock.Anything).Once().Return(nil).Run(confirmSurvey())
-
-	sm.On("AskOne", &survey.Confirm{
-		Message: "Do you want to remove service(s)' persistent data as well?",
-		Default: false,
-	}, mock.Anything, mock.Anything).Once().Return(nil)
-
-	me.On("ServiceDelete", false, serviceID).Once().Return(nil)
-
-	require.NoError(t, c.cmd.Execute())
+	flags.Set("all", "true")
+	require.True(t, c.all)
 
-	sm.AssertExpectations(t)
-	me.AssertExpectations(t)
+	flags.Set("keep-data", "true")
+	require.True(t, c.keepData)
 }
 
-func TestServiceDeleteWithServiceIDAndYesFlag(t *testing.T) {
-	var (
-		serviceID = "1"
-		sm        = newMockSurvey()
-		me        = newMockExecutor()
-		c         = newServiceDeleteCmd(me, sm)
-	)
-
-	c.cmd.SetArgs([]string{serviceID})
-	c.cmd.Flags().Set("yes", "true")
-
-	me.On("ServiceDelete", true, serviceID).Once().Return(nil)
-
-	require.NoError(t, c.cmd.Execute())
-
-	sm.AssertExpectations(t)
-	me.AssertExpectations(t)
+func TestServiceDeletePreRunE(t *testing.T) {
+	c := newServiceDeleteCmd(nil)
+	c.discardOutput()
+	require.Equal(t, errNoID, c.preRunE(c.cmd, nil))
 }
 
-func TestServiceDeleteWithAllAndYesFlags(t *testing.T) {
+func TestServiceDeleteRunE(t *testing.T) {
```

Comment:
This test doesn't cover the tests deleted within this commit. I don't understand why they're removed.

Reply:
It tests only for runE, the previous tests were for both preRunE and runE at the same time